### PR TITLE
Galgo integration

### PR DESCRIPTION
**Pull request by ElianaTroper**

### CODE REVIEW

**Comment by ElianaTroper** [comment]
@mccabe-david can you take a look at the transactions I indicated in discord?

File: src/components/Holdings.jsx [hunk 1]
```diff
@@ -28,9 +29,9 @@ function getAssets() {
   var assets = [];
   let x = getWalletInfo()["assets"];
   for (var i = 0, len = x.length; i < len; i++) {
-    if ([684649988, 684649672, 692432647].includes(x[i]["asset-id"])) {
+    if ([ids.asa.gard, ids.asa.gain, ids.asa.gardian, ids.asa.galgo].includes(x[i]["asset-id"])) {
       let amnt = (x[i]["amount"] / 10 ** x[i]["decimals"]).toFixed(3);
-      let token_price = x[i]["asset-id"] == 684649988 ? 1 : 0;
+      let token_price = x[i]["asset-id"] == ids.asa.gard ? 1 : 0;
```
Comment (mccabe-david) [inline src/components/Holdings.jsx:34]: We should assign a value to gAlgo tokens, too. 0.98 * Algo price should do it

File: src/components/Positions.jsx [hunk 2]
```diff
@@ -75,7 +88,7 @@ const dummyCDPs = [
   ];
 
   function displayRatio() {
-    return calcRatio(algosToMAlgos(getCollateral()), getMinted(), true);
+    return calcRatio(algosToMAlgos(getCollateral()), getMinted(), 0, true); // TODO: Need to set the ASA ID Properly
```
Comment (mccabe-david) [inline src/components/Positions.jsx:91]: Passing an argument to this function would accomplish the behavior we need.

File: src/components/SupplyCDP.jsx [hunk 3]
```diff
@@ -59,6 +71,13 @@ export default function SupplyCDP({
       ) / 100;
    }
 
+  //  const handleSelect = (e) => {
```
Comment (mccabe-david) [inline src/components/SupplyCDP.jsx:74]: Why not delete this bit?

File: src/transactions/cdp.js [hunk 4]
```diff
@@ -385,14 +364,161 @@ export async function openCDP(openingALGOs, openingGARD, commit, toWallet) {
...
+    };
+  }
+  */ // TODO: Fix this
+
+  let params = await paramsPromise;
+  let txns = makeOptInTxns(info, params);
+  let optins = txns.length;
+  params.fee = 7000;
+  // next two txns
+  txns = txns.concat(_openCDPtxns1(850000, cdp, info, params)) // XXX: This amount may not be optimized.
```
Comment (mccabe-david) [inline src/transactions/cdp.js:401]: If addresses are re-used. We shouldn't be doing this 850,000. Let's add a check in the function for current cdp balance?

**Review by mccabe-david — APPROVED** [review]
Tested All transactions aside from commitment and swapping. Added a few Jira bugs that were introduced, but on the whole this is functionally strong. Solid work everyone!

**Pull request merged**